java/src/main/java/org/rocksdb/CompressionType.java

```diff
@@ -81,13 +79,14 @@ public static CompressionType getCompressionType(byte byteIdentifier) {
    *
    * @return CompressionType instance (optional)
    */
-  public static Optional<CompressionType> getFromInternal(final String internalName) {
+  static CompressionType getFromInternal(final String internalName) {
     for (final CompressionType compressionType : CompressionType.values()) {
       if (compressionType.internalName_.equals(internalName)) {
-        return Optional.of(compressionType);
+        return compressionType;
       }
     }
-    return Optional.empty();
+
+    throw new IllegalArgumentException("Illegal internalName provided for CompressionType.");
```

Comment:
nit: Any reason not to put what the bad name is into the thrown exception?

Reply:
@mrambacher The Java 7 stuff shouldn't be in this PR, and so I will remove it.
I have addressed your comment above in https://github.com/facebook/rocksdb/pull/9103 (you may need to reimport that one).